main.ts

```diff
@@ -223,38 +233,47 @@ export default class ReadwiseMirror extends Plugin {
 
   async writeLibraryToMarkdown(library: Library) {
     const vault = this.app.vault;
-
+    
     // Create parent directories for all categories, if they do not exist
     library['categories'].forEach(async (category: string) => {
       category = category.charAt(0).toUpperCase() + category.slice(1); // Title Case the directory name
-
+      
       const path = `${this.settings.baseFolderName}/${category}`;
       const abstractFolder = vault.getAbstractFileByPath(path);
-
+      
       if (!abstractFolder) {
         vault.createFolder(path);
         console.info('Readwise: Successfully created folder', path);
       }
     });
-
+    
+    // Get total number of records
+    const booksTotal = Object.keys(library.books).length;
+    let bookCurrent = 1;
     for (let bookId in library['books']) {
+      this.notify.setStatusBarText(`Readwise: Processing - ${Math.floor(bookCurrent/booksTotal *100)}% finished (${bookCurrent}/${booksTotal})`);
+      bookCurrent += 1;
       const book = library['books'][bookId];
 
       const {
-        id,
+        user_book_id,
         title,
+        document_note,
+        summary,
         author,
         category,
-        num_highlights,
-        updated,
         cover_image_url,
-        highlights_url,
+        readwise_url,
         highlights,
-        last_highlight_at,
         source_url,
-        tags,
+        book_tags,
       } = book;
 
+      // Get highlight count
+      const num_highlights = book['highlights'].length;

```

Comment:
Since we're already destructuring all of the other properties out of `book`, can we also use `highlights` here instead of `book['highlights']` for these 3 items to keep everything consistent?

Reply:
good catch, implemented in the latest commit